pkg/kstatus/status/core.go

```diff
@@ -356,43 +357,94 @@ func podConditions(u *unstructured.Unstructured) (*Result, error) {
 	if err != nil {
 		return nil, err
 	}
-	phase := GetStringField(obj, ".status.phase", "unknown")
+	phase := GetStringField(obj, ".status.phase", "")
 
-	if phase == "Succeeded" {
+	switch phase {
+	case "Succeeded":
 		return &Result{
 			Status:     CurrentStatus,
 			Message:    "Pod has completed successfully",
 			Conditions: []Condition{},
 		}, nil
-	}
+	case "Failed":
+		return &Result{
+			Status:     CurrentStatus,
+			Message:    "Pod has completed, but not successfully",
+			Conditions: []Condition{},
+		}, nil
+	case "Running":
+		if hasConditionWithStatus(objc.Status.Conditions, "Ready", corev1.ConditionTrue) {
+			return &Result{
+				Status:     CurrentStatus,
+				Message:    "Pod is Ready",
+				Conditions: []Condition{},
+			}, nil
+		}
 
-	for _, c := range objc.Status.Conditions {
-		if c.Type == "Ready" {
-			if c.Status == corev1.ConditionTrue {
-				return &Result{
-					Status:     CurrentStatus,
-					Message:    "Pod has reached the ready state",
-					Conditions: []Condition{},
-				}, nil
-			}
-			if c.Status == corev1.ConditionFalse && c.Reason == "PodCompleted" && phase != "Succeeded" {
-				message := "Pod has completed, but not successfully."
-				return &Result{
-					Status:  FailedStatus,
-					Message: message,
-					Conditions: []Condition{{
-						Type:    ConditionFailed,
-						Status:  corev1.ConditionTrue,
-						Reason:  "PodFailed",
-						Message: fmt.Sprintf("Pod has completed, but not succeesfully."),
-					}},
-				}, nil
-			}
+		containerNames, isCrashLooping, err := getCrashLoopingContainers(obj)
+		if err != nil {
+			return nil, err
 		}
+		if isCrashLooping {
+			return newFailedStatus("ContainerCrashLooping",
+				fmt.Sprintf("Containers in CrashLoop state: %s", strings.Join(containerNames, ","))), nil
+		}
+
+		return newInProgressStatus("PodRunningNotReady", "Pod is running but is not Ready"), nil
```

Comment:
have you considered using enums so we can lean on the compiler and IDE to maintain these state names?

Reply:
I have thought about it. These roughly corresponds to the `Reason` field in the Conditions, which are of type string and does not have a clearly defined set of possible values. So I have kept them as strings so far.
Longer term I would like to find a way to automatically generate documentation that would enumerate the possible values here with a description of what it actually means.